### DISCUSSION TITLE
Send form data to command?

### DISCUSSION THREAD

**Discussion by cmer** [hopsoft/turbo_boost-commands#70] [Q&A]
With Stimulus Reflex, we can send the surrounding form data by setting `serializeForm=true`. Is there an equivalent for TBC? I have buttons triggering commands within a form, and the backend needs to know what the current form values are.

**Comment by hopsoft** [comment]
Commands doesn't support this; however, Commands do work with forms, so you could wire up a command to run on form submit.

**Comment by hopsoft** [comment]
Here's an example from a real world project. Note that this code may produce more questions for you, but at least you can see how it's being used in at least one real world example.

<img width="898" alt="Monosnap edit_current_dir 2023-01-12 06-21-23" src="https://user-images.githubusercontent.com/32920/212077960-8aaa4c6f-30da-43b3-b4e0-b4463d505e94.png">

**Comment by cmer** [comment]
Are there any plans to add this functionality? Not being able to access form values unless the form is submitted seems like a major downside. For my particular use-case at least.

**Comment by hopsoft** [comment]
Most likely won't be implemented as a feature in the sense that you're thinking. However, you can accomplish what you need in other ways. For example, you can listen for the `turbo-boost:command:start` event and update the triggering element with any additional attributes you need. 

Though, I don't think we'll ever make collecting all form attributes be implicit behavior in the core library.

**Reply by hopsoft** [reply]
Note that this event is triggered on the element and bubbles up to window.

**Comment by cmer** [comment]
I'd love to see this behaviour added:

`data-turbo-command-serialize-form="true"` serializes the closest form

and

`data-turbo-command-serialize-form="#any_id"` serializes the form fields within the selector. The selector could be the form itself, or a subset of a form.

I'm toying with TBC and trying to see how I could re-implement [Invoice Beast](https://github.com/cmer/stimulus-reflex-invoice-beast) with it. Form serialization whenever an action is triggered is needed.

**Comment by julianrubisch** [comment]
I agree with the general sentiment, but we want to keep the core library much slimmer than before.

Note that Rails provides perfectly sensible form handling out of the box that with the advent of Turbo gained reactivity. Most of the time, a simple `button_to` will fulfill all of your requirements.

Note furthermore that SR‘s `serializeForm` behavior stems from _before_ the Turbo era and has been criticized as overreaching afterwards.
